src/arrays/programs_involving_data_sequences/hard/q7/MainTest.java

```diff
@@ -0,0 +1,96 @@
+
+public class MainTest extends BaseTest {
+    // Java
+
+    public static int n = 10;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = arrayToInput(generateRandomArray(-10, 10, n));
+        return new Clause[]{
+                new StringLiteral("Enter an array of " + n + " integers to be squared:"),
+                new Optional(new StringLiteral(" ")),
+                new NewLine(),
+                new StringLiteral("The squared array is:"),
+                new NewLine(),
+                new PlaceHolder(),
+                new Optional(new StringLiteral(" "))
+        };
+    }
+
+    public void runMain() {
+        SquareUp.main(new String[0]);
+    }
+
+    public static int[] sq(int[] arr) {
+        int[] input = arr.clone();
+        for (int i = 0; i < input.length; i++)
+            input[i] = (int) Math.pow(input[i], 2);
+        return input;
+    }
+
+    static Stream<Arguments> mainInputProvider() {
+        return Stream.of(
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-10000, 10000, n)),
+                Arguments.of(generateRandomArray(-1, 2, n)),
+                Arguments.of(replicateArray(1, n)),
+                Arguments.of(replicateArray(0, n)),
+                Arguments.of(replicateArray(10000, n))
+
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainInputProvider")
+    void printCorrectOutput(int[] a) throws InvalidClauseException {
+        TestOption.incorrectStructureErrorMessage = "Your program does not print the correct sequence of squared array.";
+        runWithInput(arrayToInput(a), new Clause[]{
+                new StringLiteral(arrayToInput(sq(a)))
+        });
+    }
+
+    static Stream<Arguments> methodInputProvider() {
+        return Stream.of(
+                Arguments.of(generateRandomArray(-10000, 10000, 10000)),
+                Arguments.of(generateRandomArray(-10000, 10000, 10000)),
+                Arguments.of(generateRandomArray(-10000, 10000, 1000)),
+                Arguments.of(generateRandomArray(-1, 2, 10000)),
+                Arguments.of(replicateArray(1, 10000)),
+                Arguments.of(replicateArray(0, 10000)),
+                Arguments.of(replicateArray(10000, 10000))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("methodInputProvider")
+    void correctFibonacciMaker(int[] input) throws Throwable {
+        Object[][] arguments = {
+                {input, int[].class},
+        };
+        MethodTest m = new MethodTest(SquareUp.class, "doubler", arguments);
+        m.setIncorrectMethodStructureErrorMessage("Your doubler method does not return the correct array of squared integers.");
+        Object output = m.callMethod();
+        assertArrayEquals((int[]) output, sq(input));
```

Comment:
Please calculate the expected output before you call the method in case the student does not create a new array and passes back the original one instead. Also, move the expected value before the output, use CustomAssertions, and make you incorrect structure error a test error.